javascript/javascript1/week1/homework_1.sql

```diff
@@ -0,0 +1,62 @@
+--1. Find out how many tasks are in the task table
+
+SELECT COUNT (*)
+FROM task
+
+--2. Find out how many tasks in the task table do not have a valid due date
+
+SELECT COUNT (*)
+FROM task 
+WHERE due_date IS NULL
+
+--3. Find all the tasks that are marked as done
+
+SELECT task.title, status.name
+FROM task JOIN status ON task.status_id = status.id
+WHERE status.name IN ('Done')
+
+--4. Find all the tasks that are not marked as done
+
+SELECT task.title, status.name
+FROM task JOIN status ON task.status_id = status.id
+WHERE status.name NOT IN ('Done')
+
+--5. Get all the tasks, sorted with the most recently created first
+
+SELECT * 
+FROM task
+ORDER BY created
+
+--6. Get the single most recently created task
+
+SELECT * 
+FROM task
+ORDER BY created
+LIMIT 1 
+
+--7. Get the title and due date of all tasks where the title or description contains database
+
+SELECT task.title, task.due_date
+FROM task 
+WHERE task.title LIKE '%database%' 
+   OR task.description LIKE '%database%';
+
+--8. Get the title and status (as text) of all tasks
+
+SELECT task.title, status.name
+FROM task JOIN status ON task.status_id = status.id
```

Comment:
It would have been nice to give the `status.name` field an alias with `AS` keyword. This way it shows up in the result as `name`, even though it represents a status.